glean-core/rlb/src/test.rs

```diff
@@ -118,20 +180,53 @@ fn queued_recorded_metrics_correctly_record_during_init() {
 
 #[test]
 fn initializing_twice_is_a_noop() {
+    let _lock = GLOBAL_LOCK.lock().unwrap();
+    env_logger::try_init().ok();
+
     let dir = tempfile::tempdir().unwrap();
     let tmpname = dir.path().display().to_string();
 
-    let cfg = Configuration {
-        data_path: tmpname,
-        application_id: GLOBAL_APPLICATION_ID.into(),
-        upload_enabled: true,
-        max_events: None,
-        delay_ping_lifetime_io: false,
-        channel: Some("testing".into()),
-    };
-
-    initialize(cfg.clone(), ClientInfoMetrics::unknown());
-    initialize(cfg, ClientInfoMetrics::unknown());
+    reset_glean(
+        Configuration {
+            data_path: tmpname.clone(),
+            application_id: GLOBAL_APPLICATION_ID.into(),
+            upload_enabled: true,
+            max_events: None,
+            delay_ping_lifetime_io: false,
+            channel: Some("testing".into()),
+            server_endpoint: Some("invalid-test-host".into()),
+            uploader: None,
+        },
+        ClientInfoMetrics::unknown(),
+        true,
+    );
+
+    // Glean init is async and we need to wait for it to complete. There
+    // is nothing we can do but wait. Tests in other bindings use
```

Comment:
Waiting for the queue to unblock?

Reply:
Right, I think we can do this here as well, given your latest fix to the dispatcher. Looks like I wasn't able to due to the `block_on_queue` waiting on an old dispatcher :)